script/src/verify.rs

```diff
@@ -76,30 +76,29 @@ impl<'a, DL: DataLoader> TransactionScriptsVerifier<'a, DL> {
         let resolved_inputs = &rtx.resolved_inputs;
         let outputs = rtx
             .transaction
-            .outputs()
-            .iter()
+            .outputs_with_data_iter()
             .enumerate()
-            .map(|(index, output)| CellMeta {
-                cell_output: Some(output.clone()),
+            .map(|(index, (output, data))| CellMeta {
+                cell_output: output.to_owned(),
                 out_point: CellOutPoint {
                     tx_hash: tx_hash.to_owned(),
                     index: index as u32,
                 },
                 block_info: None,
                 cellbase: false,
-                capacity: output.capacity,
-                data_hash: None,
+                data_bytes: data.len() as u32,
+                mem_cell_data: Some(data.to_owned()),
             })
             .collect();
 
         let mut binaries_by_data_hash: FnvHashMap<H256, Bytes> = FnvHashMap::default();
         let mut binaries_by_type_hash: FnvHashMap<H256, Bytes> = FnvHashMap::default();
         for resolved_dep in resolved_deps {
             if let Some(cell_meta) = &resolved_dep.cell() {
-                let output = data_loader.lazy_load_cell_output(cell_meta);
-                binaries_by_data_hash.insert(output.data_hash(), output.data.to_owned());
-                if let Some(t) = &output.type_ {
-                    binaries_by_type_hash.insert(t.hash(), output.data.to_owned());
+                let data = data_loader.load_cell_data(cell_meta).expect("cell data");
+                binaries_by_data_hash.insert(cell_meta.data_hash().to_owned(), data.to_owned());
+                if let Some(t) = &cell_meta.cell_output.type_ {
+                    binaries_by_type_hash.insert(t.hash(), data.to_owned());
```

Comment:
I think we can move this part of code to `extract_script` now. @xxuejie

Reply:
I think we should be all good here.